CardPayments/src/main/java/com/paypal/android/cardpayments/CardApproveOrderCallback.kt

```diff
@@ -0,0 +1,13 @@
+package com.paypal.android.cardpayments
+
+import androidx.annotation.MainThread
+
+fun interface CardApproveOrderCallback {
+    /**
+     * Called when the order is approved.
+     *
+     * @param result [CardApproveOrderResult] result with details
+     */
+    @MainThread
+    fun onCardApproveOrderResult(result: CardApproveOrderResult)
```

Comment:
That's interesting. I don't think it's possible to do this in iOS with protocols.
Do you feel like this is something merchant would want us to handle?

Reply:
I'm not sure if we need to deliver results on the main thread, but it is convenient when updating the UI after executing a transaction.

It's up to y'all if you want to do the same. I feel like the closest equivalent to this annotation in Swift would be `@MainActor`.